src/main/java/org/snomed/snowstorm/core/data/services/RelationshipService.java

```diff
@@ -46,9 +52,49 @@
 	@Autowired
 	private ConceptUpdateHelper conceptUpdateHelper;
 
+	@Autowired
+	private MRCMLoader mrcmLoader;
+
 	public Relationship findRelationship(String branchPath, String relationshipId) {
```

Comment:
This method will be used when fetching a single relationship so this API endpoint probably works: `GET /{branch}/relationships/{relationshipId}`
We also need the concrete types set when viewing the inferred relationships within a concept using `GET /browser/{branch}/concepts/{conceptId}` and viewing the list under `{"relationships" : [] }`. 
The setConcreteValueFromMRCM method may have to become protected or public and allow a collection of relationships to be passed in. I think it would be good separation of concerns for the MRCMLoader to have a method like getConcreteType(attributeType) so the relationship service is doing less work. The data type is an MRCM concern really. The AttributeRange could have a field for the data type which gets populated on load.

Reply:
Thank you for picking up on that @kaicode. 

I have implemented both suggestions.